src/python/WMCore/JobStateMachine/ChangeState.py

```diff
@@ -302,7 +305,20 @@ def recordInCouch(self, jobs, newstate, oldstate, updatesummary = False):
                 logging.debug("Updated job summary state history for job %s" % jobSummaryId)
 
             if job.get("fwjr", None):
-
+                cached_workflow = self.workflow_cache.setdefault(job['workflow'], {})
+                if job['task'] not in cached_workflow:
+                    workload = WMWorkloadHelper()
+                    workload.load(self.getWorkflowSpecDAO.execute(job['task'])['spec'])
+                    cached_workflow['campaign'] = workload.getCampaign()
+                    for task in workload.taskIterator():
+                        cached_workflow[task.getPathName()] = task.getPrepID()
+                    if job['task'] not in cached_workflow:
+                        logging.debug("Couldn't find the task in the cache after "
+                                      "it should have been added!")
+
+                fwjr = job['fwjr']
+                fwjr.setCampaign(cached_workflow.get('campaign', 'Unknown'))
+                fwjr.setPrepID(cached_workflow.get(job['task'], 'Unknown'))
```

Comment:
remove this line when 
https://github.com/alexanderrichards/WMCore/blob/748cf8d19dbd2b6094c68e2df694c59cc122a7c0/src/python/WMCore/FwkJobReport/Report.py#L1329
is modified.

Also updates the comments (it seems it is copied from setGlobalTag())

I am thinking whether we should do the same thing for setAcquisitionProcessing and setGlobalTag  in case we need to have the statistics by GlobalTag or Acquisition era? @amaltaro, Alan what do you think?